vk-video/README.md

```diff
@@ -0,0 +1,91 @@
+
+## Usage
+
+```rs
+fn decode_video(
+    window: &winit::window::Window,
+    mut encoded_video_reader: impl std::io::Read,
+) {
+    let instance = vk_video::VulkanInstance::new().unwrap();
+    let mut surface = instance.wgpu_instance.create_surface(window).unwrap();
+    let device = instance
+        .create_device(
+            wgpu::Features::empty(),
+            wgpu::Limits::default(),
+            &mut Some(&mut surface),
+        )
+        .unwrap();
+
+    let mut decoder = device.create_wgpu_textures_decoder().unwrap();
+    let mut buffer = vec![0; 4096];
+
+    while let Ok(n) = encoded_video_reader.read(&mut buffer) {
+        if n == 0 {
+            return;
+        }
+
+        let decoded_frames = decoder.decode(&buffer[..n], None).unwrap();
+
+        for frame in decoded_frames {
+            // Each frame contains a wgpu::Texture you can sample for drawing.
+            // device.wgpu_device is a wgpu::Device and device.wgpu_queue
+            // is a wgpu::Queue. You can use these for interacting with the frames.
+        }
+    }
+}
+```
+Be sure to check out our examples, especially the `player` example, which is a simple video player built using this library and wgpu. Because the player is very simple, you need to extract the raw h264 data from a container before usage. Here's an example on how to extract the h264 bytestream out of an mp4 file using ffmpeg:
+
+```sh
+ffmpeg -i input.mp4 -c:v copy -bsf:v h264_mp4toannexb -an output.h264
+```
+
+Then you can run the example with:
+
+```sh
+git clone https://github.com/software-mansion/live-compositor.git
+cd live-compositor/vk-video
+cargo run --example player -- output.h264 FRAMERATE
+```
+
+## Compatibility
+
+On linux, the library should work on NVIDIA GPUs out of the box. For AMD GPUs with recent Mesa drivers, you need to set the `RADV_PERFTEST` environment variable to `video_decode` for now:
+
+```sh
+RADV_PERFTEST=video_decode cargo run
+```
+
+It should work on Windows with recent drivers out of the box. Be sure to submit an issue if it doesn't.
+
+[wgpu]: https://wgpu.rs/
+[`wgpu::Texture`]: https://docs.rs/wgpu/latest/wgpu/struct.Texture.html
+
+## vk-video is created by Software Mansion
+
+[![swm](https://logo.swmansion.com/logo?color=white&variant=desktop&width=150&tag=live-compositor-github 'Software Mansion')](https://swmansion.com)
+
+Since 2012 [Software Mansion](https://swmansion.com) is a software agency with experience in building web and mobile apps as well as complex multimedia solutions. We are Core React Native Contributors and experts in live streaming and broadcasting technologies. We can help you build your next dream product – [Hire us](https://swmansion.com/contact/projects?utm_source=live-compositor&utm_medium=readme).
```

Comment:
```suggestion
Since 2012 [Software Mansion](https://swmansion.com) is a software agency with experience in building web and mobile apps as well as complex multimedia solutions. We are Core React Native Contributors and experts in live streaming and broadcasting technologies. We can help you build your next dream product – [Hire us](https://swmansion.com/contact/projects?utm_source=live-compositor-vk-video&utm_medium=readme).
```